### PR TITLE
Expiring emails

### LINKED ISSUE #168: Expiring emails for voting
- [x] Invitation emails should contain a link to a pre-vote page.
- [x] On the pre-vote page, when voters enter an email, an email should be sent to them with an expiring-token that takes them to a page where they can vote
- [x] pre-vote page checks if user is eligible to vote
- [x] The remote vote page will encode all artworks (instead of only those selected/scanned)
- [x] The pre-vote page should contain copy explaining why the vote page will request location.
- [x] Must remove the invitation and redis key after user has voted
- [x] Handle error if invitation key has expired
- [x] Test for invitation endpoint
- [x] Test for new task kind

### PR DESCRIPTION
**Pull request by ana0**
Close #168 

The basic flow that this PR adds is that the expiring vote emails. When uploading via the schedule emails page, a link is sent to each voter that directs them to a "pre-vote page" - on the pre-vote page, they can enter their email. 

If there is a vote invitation associated with this email, they are sent another email, with a vote link that will expire in 15 minutes. If clicked, it sends them to the vote page. If clicked a second time, it no longer works.

If the user requests this link but doesn't follow through to voting, they can request a new, unique, expiring link. If they did finish voting, they can no longer request an expiring vote link

Steps to test this:

1) make a vote with all needed components (festival, artwork, artist, festival question, artwork question, etc)
2) use the schedule emails page to create an invitation for at least one email
3) navigate the link logged in the sent emails, and input the email in the form, which will log a new (expiring) email job
4) navigate to the link logged in the new expiring email job, where you should be able to successfully vote

### CODE REVIEW

**Comment by critocrito** [comment]
Hey @ana0, thanks a lot for this. It works great for me. I tried to setup several votes and invited myself to participate. All working out as promised. I did find a few nitpicking things, let me know what you think of those.

**Comment by ana0** [comment]
Hmm not sure I can see the changes, I only see the merge commit!

**Comment by critocrito** [comment]
sorry, forgot to press the button ...

File: src/server/validations/tasks.js [hunk 1]
```diff
@@ -23,7 +24,13 @@ const defaultValidation = {
           )
           .min(1)
           .required(),
-        otherwise: Joi.object().allow(null).default(null),
```
[the remark this answers is not in the document]
Comment (ana0) [reply src/server/validations/tasks.js:26]: Pretty sure it throws a bad request, for example ["should fail with an unknown task type"](https://github.com/bitspossessed/culturestake/blob/129a7914bb11ac86cd8ee3a0eac107ee0e9fefa6/test/tasks.test.js#L35) is still passing

File: src/client/views/RemoteVote.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+  useEffect(() => {
+    return () => {
+      dispatch(resetVote());
+    };
+  }, [dispatch]);
+
+  return (
+    <Fragment>
+      <View>
+        {isVoteReady && !isInvitationLoading & !isError ? (
```
[the remark this answers is not in the document]
Comment (ana0) [reply src/client/views/RemoteVote.js:67]: Definitely! Thanks for spotting

**Pull request merged**